src/System.CommandLine/Binding/BindingContext.cs

```diff
@@ -68,31 +70,31 @@ public void AddService<T>(Func<IServiceProvider, T> factory)
                 throw new ArgumentNullException(nameof(factory));
             }
 
-            ServiceProvider.AddService(typeof(T), s => factory(s));
+            ServiceProvider.AddService(typeof(T), s => factory(s)!);
```

Comment:
Why are we asserting not null from the factory? Wouldn't throwing be better?

Reply:
Actually it's allowed to be null. Fixing.